tree/tree/src/TChain.cxx

```diff
@@ -2303,6 +2304,47 @@ void TChain::ResetAfterMerge(TFileMergeInfo *info)
    TTree::ResetAfterMerge(info);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save TChain as a C++ statement(s) on output stream out
+
+void TChain::SavePrimitive(std::ostream &out, Option_t *option)
+{
+   TList chains;
+   chains.Add(this);
+
+   out << "   TChain *" << fName.Data() << " = new TChain(\"" << fName.Data() << "\");" << std::endl;
+
+   for (TObject *el : *fFriends) {
+      TChain *ch = (TChain *)((TFriendElement *)el)->GetTree();
```

Comment:
The friend of a chain is not necessarily a TChain, it could also be a TTree.

Reply:
Probably then TTree::SavePrimitive() should be realized with reading it from a ROOT file. TChain::SavePrimitive() would just call it for TTree friends. What do you think?

Reply:
Or I'd just check ClassName and store friends only for TChain friends.

Reply:
It is not clear what TTree::SavePrimitives 'should' do.  There is two options.   If doing the same as others SavePrimitive it would printing its contents which could 'just' be the structure (i.e. reforming the branches) or could even be the structure and the data (but then the resulting file could be huge).   The other alternative, more inline with the TChain::SavePrimitive would be store open the TFile and grap the TTree from it.

Because of this potential confusion I hesitate to have an official TTree::SavePrimitives but TChain's could handle the TTree using the "other/last alternative" directly.